### PR TITLE
Make .devcontainer language specific

### LINKED ISSUE #88: Audit devcontainer.json per template
There are likely a couple ways that we can improve our devcontainer.json.

One example is that right now, we don't have key language extensions (e.g. `ms-python.python`) included in the devcontainer.json for Python templates but we do have `ms-dotnettools.csharp` and `ms-dotnettools.vscode-dotnet-runtime` which are unnecessary . Ideally, we should tailor the devcontainer.json so that the extensions support/complement the template language/technologies.

### PR DESCRIPTION
**Pull request by savannahostrowski**
Closes #88 

There was a comment on this issue about deleting the reference to common/.devcontainer from repo.yaml but I don't see that in the file. Let me know if I missed anything!

### CODE REVIEW

File: templates/todo/projects/csharp-mongo/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ARG VARIANT=bullseye
```
Comment (jongio) [inline templates/todo/projects/csharp-mongo/.devcontainer/Dockerfile:1]: Seems like this is still the same for all templates.  We could keep it in common.

File: templates/todo/projects/python-mongo-aca/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (jongio) [inline templates/todo/projects/python-mongo-aca/.devcontainer/devcontainer.json:1]: Some of these devcontainer.json files are the same.  We can re-arrange how we store them in common and then copy any that are the same out to the various templates.

**Review by jongio — CHANGES_REQUESTED** [review]

File: templates/todo/projects/csharp-mongo/.devcontainer/Dockerfile (hunk 1 above)
Comment (jongio) [reply templates/todo/projects/csharp-mongo/.devcontainer/Dockerfile]: Fixed in my commit

File: templates/todo/projects/python-mongo-aca/.devcontainer/devcontainer.json (hunk 2 above)
Comment (jongio) [reply templates/todo/projects/python-mongo-aca/.devcontainer/devcontainer.json]: Fixed in my commit

**Review by jongio — APPROVED** [review]

**Comment by jongio** [comment]
Marking as draft so we can do a manual test pass against all of these templates with the new devcontainer updates.

File: templates/common/.devcontainer/Dockerfile/base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ARG VARIANT=bullseye
+FROM mcr.microsoft.com/vscode/devcontainers/base:0-${VARIANT}
+RUN curl -fsSL https://aka.ms/install-azd.sh | bash
```
Comment (vhvb1989) [inline templates/common/.devcontainer/Dockerfile/base/Dockerfile:3]:
```suggestion
RUN curl -fsSL https://aka.ms/install-azd.sh | bash

```

**Review by vhvb1989 — APPROVED** [review]
It would be good to have CI/CD tests to validate the Dockerfile (can build OK) and devcontainer is valid

**Comment by jongio** [comment]
> It would be good to have CI/CD tests to validate the Dockerfile (can build OK) and devcontainer is valid

It is now possible https://github.com/Azure/azure-dev/issues/504

**Pull request merged**